confidant/core/secrets.py

```
@@ -0,0 +1,120 @@
+    """
+    Encrypt a plain text secret using the master key.
+
+    Example:
+        encrypt_secret("my-secret") -> "ENC(...)"
+    """
+    try:
+        fernet = Fernet(_get_master_key())
+        encrypted = fernet.encrypt(plain_text.encode()).decode()
+        return f"ENC({encrypted})"
+    except Exception as e:
+        raise SecretsError(f"Encryption failed: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)